nats.go

```diff
@@ -3090,6 +3093,29 @@ func (nc *Conn) respHandler(m *Msg) {
 	}
 }
 
+// CustomInboxPrefix is an Option to set the inbox prefix.
+func CustomInboxPrefix(inboxPrefix string) Option {
```

Comment:
@wallyqs I think this validation should be in opts.Connect(), since the public field CustomInboxPrefix may be modified externally